java/ql/src/semmle/code/java/frameworks/spring/SpringWebUtil.qll

```diff
@@ -0,0 +1,241 @@
+        "org.springframework.web.util;CookieGenerator;false;addCookie;;;Argument[1];Argument[0];taint",
+        "org.springframework.web.util;CookieGenerator;false;getCookieDomain;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;CookieGenerator;false;setCookieDomain;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;CookieGenerator;false;getCookieName;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;CookieGenerator;false;setCookieName;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;CookieGenerator;false;getCookiePath;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;CookieGenerator;false;setCookiePath;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;DefaultUriBuilderFactory;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;builder;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;expand;;;Argument[-1..0];ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;expand;(String,Map);;MapValue of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;expand;(String,Object[]);;ArrayElement of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;getDefaultUriVariables;;;Argument[-1];MapValue of ReturnValue;taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;setDefaultUriVariables;;;MapValue of Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;DefaultUriBuilderFactory;false;uriString;;;Argument[-1];ReturnValue;taint",
+        // note for review: I couldn't find any default xss sanitizer definitions so I've assumed I shouldn't model html escape functions for taint
+        "org.springframework.web.util;HtmlUtils;false;htmlUnescape;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;ServletContextPropertyUtils;false;resolvePlaceholders;;;Argument[0..1];ReturnValue;taint",
+        "org.springframework.web.util;ServletRequestPathUtils;false;getCachedPath;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;ServletRequestPathUtils;false;getCachedPathValue;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;ServletRequestPathUtils;false;getParsedRequestPath;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;ServletRequestPathUtils;false;parseAndCache;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;ServletRequestPathUtils;false;setParsedRequestPath;;;Argument[0];Argument[1];taint",
+        // for review: this is likely a good use for synthetic fields; should I use them and wait for that PR to be merged?
```

Comment:
Support for synthetic fields is merged, so if they help here, then that's an option.